addons/im_livechat/models/res_partner.py

```diff
@@ -94,7 +94,9 @@ def _compute_display_name(self):
         if not self.env.context.get("im_livechat.hide_partner_company"):
             super()._compute_display_name()
             return
-        for partner in self:
+        portal_partners = self.filtered("partner_share")
+        super(ResPartner, portal_partners)._compute_display_name()
+        for partner in self - portal_partners:
```

Comment:
We might want to manually add the prefetch to both recordsets after filtering as they are likely to need similar fields. We don't want to fetch fields for only portal_partners and then re-fetch the same fields for self - portal_partners

Reply:
But maybe by "luck" filtering on `partner_share` will already prefetch the whole recordset at once. So it might be ok for now.

Reply:
Ok for now then. The lack of common prefetch would be an issue if both split recordsets needed to follow the same deeper relation, as this would not be prefetched initially.